scene/gui/tab_container.cpp

```diff
@@ -611,6 +611,11 @@ void TabContainer::set_current_tab(int p_current) {
 		setup_current_tab = p_current;
 		return;
 	}
+	// Discard setup if the current tab was changed right after entering the tree.
+	if (setup_current_tab > -2) {
+		setup_current_tab = -2;
+	}
+
```

Comment:
I did some test : move `set_current_tab(setup_current_tab)` from `NOTIFICATION_POST_ENTER_TREE` to `NOTIFICATION_ENTER_TREE` also fix this bug, not sure why it has to be `NOTIFICATION_POST_ENTER_TREE`